src/Functions/IFunction.cpp

```diff
@@ -179,10 +178,8 @@ NullPresence getNullPresense(const ColumnsWithTypeAndName & args)
 
     for (const auto & elem : args)
     {
-        if (!res.has_nullable)
-            res.has_nullable = elem.type->isNullable();
-        if (!res.has_null_constant)
-            res.has_null_constant = elem.type->onlyNull();
+        res.has_nullable |= elem.type->isNullable();
```

Comment:
Is this really an optimization? Now it has to read the old value, while before it only wrote.

Reply:
Oops, commented wrong line -- I meant line 151 and the old value of `result_null_map[i]`.

Reply:
@akuzm it is sequential loop thought array it will be in cache. But this branch if (src_null_map[I]) compiler cannot eliminate.